src/node.js

```diff
@@ -22,6 +22,44 @@
 
     EventEmitter.call(process);
 
+    // Default process warning handler..
+    process.on('warning', (warning) => {
+      // By default, print the short warning message if --no-warnings is not
+      // set. If --trace-warnings is set, then the stack trace for the warning
+      // is shown. This takes backwards compatibility with --trace-deprecation
+      // into consideration also.
+      if (process.noProcessWarnings) return;
+      const trace = process.traceProcessWarnings ||
+                    (warning.name === 'DeprecationWarning' &&
+                     process.traceDeprecation);
+      if (trace)
+        console.error(warning.stack);
+      else
+        console.error(`${warning.name}: ${warning.message}`);
+    });
+
+    // process.emitWarning(error)
+    // process.emitWarning(str)
+    // process.emitWarning(options)
+    var Warning;
+    process.emitWarning = function(warning) {
```

Comment:
An alternative is to only accept a string or error object, along with an optional name

`(warning[, name])` 

and to default to 'Warning'... another point is that if you pass in an error object, you will want to override the name property if its 'Error'


All of this logic may be better served in the Warning constructor though.